src/components/vuestic-components/va-form/VaForm.vue

```diff
@@ -0,0 +1,162 @@
+<script>
+import { ContextPluginMixin, getContextPropValue } from '../../context-test/context-provide/ContextPlugin'
+
+const availableFormTags = ['VaInput', 'VaColorInput', 'VaDatePicker', 'VaCheckbox', 'VaRadio', 'VaSelect', 'VaFileUpload']
+const getAllChildren = (vm, children = []) => {
+  vm.$children.forEach(function (child) {
+    children.push(child)
+    child.$children.length > 0 && getAllChildren(child, children)
+  })
+  return children
+}
+
+export default {
+  name: 'VaForm',
+  mixins: [ContextPluginMixin],
+  props: {
+    lazyValidation: {
+      type: Boolean,
+      default () {
+        return getContextPropValue(this, 'lazyValidation', false)
+      },
+    },
+    autofocus: {
+      type: Boolean,
+      default () {
+        return getContextPropValue(this, 'autofocus', false)
+      },
+    },
+    tag: {
+      type: String,
+      default () {
+        return getContextPropValue(this, 'tag', 'div')
+      },
+    },
+  },
+  data () {
+    return {
+      valid: true,
+      hasResetButton: false,
+      hasSubmitButton: false,
+    }
+  },
+  mounted () {
+    this.$el.addEventListener('focusin', this.focus)
```

Comment:
Why not handle this in template as we do with normal components?

Reply:
Focusable components can be wrapped in another component. We need handling focus event on any nesting level

Reply:
@Kreezag 

How is this solution different from `@focusin` on `form` element?

Reply:
I think to an interface as `focus()` method on the parent form is better than `@focusin`

Reply:
As discussed in chat, focusin can go away. Native event is good enough.

Reply:
removed